### PR TITLE
ci: adding shellcheck

### PR DESCRIPTION
**Pull request by gauravsitlani**
Adding shellcheck github action to this repository
Signed-off-by: gauravsitlani <gauravsitlani@riseup.net>

### CODE REVIEW

File: .github/workflows/shellcheck.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    steps:
+    - uses: actions/checkout@v3
+    - name: Run ShellCheck
+      uses: ludeeus/action-shellcheck@master
+      with:
+        severity: warning
+        check_together: 'yes'
+        disable_matcher: false
+        additional_files: build/reset build/sed-in-place
+        ignore: olm
```
Comment (galexrt) [inline .github/workflows/shellcheck.yaml:33]: Let's also ignore the shellcheck error `SC2034` as might be a bit broken with some scripts that are either called from Makefiles or directly, see shellcheck github actions check output.

**Review by galexrt — CHANGES_REQUESTED** [review]

File: .github/workflows/shellcheck.yaml (hunk 1 above)
Comment (gauravsitlani) [reply .github/workflows/shellcheck.yaml]: sure

**Comment by gauravsitlani** [comment]
I believe shellcheck is working properly now.

**Review by galexrt — CHANGES_REQUESTED** [review]
Shellcheck Check action throws a warning:

> Warning: Input 'ignore' has been deprecated with message: Use ignore_paths or ignore_names instead.

Let's make sure we use the latest input cars

**Comment by gauravsitlani** [comment]
@galexrt based on the current settings, i don't see any warning now.

**Review by galexrt — APPROVED** [review]

File: .github/workflows/shellcheck.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    - name: Run ShellCheck
+      uses: ludeeus/action-shellcheck@master
+      env:
+        SHELLCHECK_OPTS: -e SC2034
+      with:
+        severity: warning
+        check_together: 'yes'
+        disable_matcher: false
+        additional_files: build/reset build/sed-in-place
+        format: gcc
```
Comment (ideepika) [inline .github/workflows/shellcheck.yaml:35]: maybe having a newline would be good here

**Review by ideepika — APPROVED** [review]
minor nit, rest lgtm!

File: .github/workflows/shellcheck.yaml (hunk 2 above)
Comment (gauravsitlani) [reply .github/workflows/shellcheck.yaml]: @ideepika sure done.

**Pull request merged**